ProteoformSuiteInternal/TheoreticalProteoform.cs

```diff
@@ -46,22 +48,57 @@ public TheoreticalProteoform(string accession, string description, string sequen
             this.ptm_set = ptm_set;
             this.unmodified_mass = unmodified_mass;
             if (check_contaminants) this.contaminant = theoretical_proteins.Where(item => item.Key.ContaminantDB).SelectMany(kv => kv.Value).Any(p => p.Accession == this.accession.Split(new char[] { '_' })[0]);
+            this.modified_mass = CalculateProteoformMass(sequence, ptm_set.ptm_combination);
         }
 
         #endregion Public Constructor
 
         #region Public Method
 
-        public static double CalculateProteoformMass(string pForm, Dictionary<char, double> aaIsotopeMassList)
+        public static double CalculateProteoformMass(string sequence, List<Ptm> ptm_combination)
         {
-            double proteoformMass = 18.010565; // start with water
-            char[] aminoAcids = pForm.ToCharArray();
-            List<double> aaMasses = new List<double>();
-            for (int i = 0; i < pForm.Length; i++)
+            if (Sweet.lollipop.theoretical_database.aaIsotopeMassList == null)
             {
-                if (aaIsotopeMassList.ContainsKey(aminoAcids[i])) aaMasses.Add(aaIsotopeMassList[aminoAcids[i]]);
+                Sweet.lollipop.theoretical_database.populate_aa_mass_dictionary();
             }
-            return proteoformMass + aaMasses.Sum();
+
+            if (!Sweet.lollipop.most_abundant_mass)
```

Comment:
It will set the modified mass as the most abundant adn just use that for everyting - for cases where deconvolution is reporting the most abundant mass for non-isotopically resolved data we have from sSEC